components/calendar/src/islamic.rs

```diff
@@ -1030,40 +1130,29 @@ impl<A: AsCalendar<Calendar = IslamicCivil>> DateTime<A> {
 pub struct IslamicTabularDateInner(ArithmeticDate<IslamicTabular>);
 
 impl CalendarArithmetic for IslamicTabular {
-    type YearInfo = ();
-
-    fn month_days(year: i32, month: u8, _data: ()) -> u8 {
-        match month {
-            1 | 3 | 5 | 7 | 9 | 11 => 30,
-            2 | 4 | 6 | 8 | 10 => 29,
-            12 if Self::is_leap_year(year, ()) => 30,
-            12 => 29,
-            _ => 0,
-        }
+    type YearInfo = IslamicYearInfo;
+
+    fn month_days(_year: i32, month: u8, year_info: IslamicYearInfo) -> u8 {
+        year_info.packed_data.days_in_month(month)
```

Comment:
Same in this class: why are you changing a straightforward match into a data-driven function?